### ISSUE TITLE
 x is not member of x in generated files error after using selectionSetInitializers

### ISSUE THREAD

**Issue by iAmericanBoy** [apollographql/apollo-ios#3112]
### Summary

After adding the `selectionSetInitializers:` option to our `OutputOptions` function the generated code is not building anymore. it looks like apollo is maybe duplicating the asUserPie inline fragment? 

Type 'ApolloSimpleSliceable.AsPie.AsUserPie' has no member 'AsUserPie' is the error we are getting in Xcode.

<img width="1421" alt="Screenshot 2023-07-11 at 9 56 50 AM" src="https://github.com/apollographql/apollo-ios/assets/32912858/98617011-ce3e-4f18-8dd7-df9322301dc6">

the part of the query that is generated by this looks like this:
``` graphql
...
  ... on Pie {
    containsCrypto
    ... on UserPie {
      id
    }
...
```

### Version

1.3

### Steps to reproduce the behavior

add selectionSetInitializers selection set to the options initializer and run `ApolloCodegen.build` the generated code now has failures in it being unable to resolve the inline fragment

### Anything else?

this can be reproduced with the same schema and fragment files I send @AnthonyMDev last week

**Comment by AnthonyMDev** [comment]
Thanks for the bug report! We're going to look into this as soon as we can. Probably will be next week, as we have some people out on vacation this week!

**Comment by AnthonyMDev** [comment]
@iAmericanBoy in order to figure out what's going on here, I am probably going to need a reproduction case. Can you provide me with an example of your schema and operation that causes this bug when generated? It doesn't need to be your entire schema, but if you can pull out just enough of the schema to make something that reproduces this issue, we'll be able to get a fix out much quicker!

**Comment by AnthonyMDev** [comment]
Whoooooops, missed the part that you said this can be reproduced with the example you already sent me last week! Disregard that last comment. We'll look into this!

**Comment by iAmericanBoy** [comment]
yup sorry I only left it as a note bit yes you can reproduce it by setting `selectionSetInitializers:[.all]` in the codegen function. you'll see the error in the `ApolloSimpleSliceable` generated code file

**Comment by calvincestari** [comment]
I think we've got another report of this issue in `1.3` over in the community forums - https://community.apollographql.com/t/apollo-v1-3-swift-ios-how-to-access-type-uploadfile/6465.

**Comment by iAmericanBoy** [comment]
Just checking in here, have you been able to reproduce this? I know the other issue is still being investigated but this is blocking us from completing the 1.x migration. Please let me know if I can help in any way? Maybe there's some investigation I could help with?

**Comment by AnthonyMDev** [comment]
I just figured out and got a fix up for the _other_ issue you were having. This one I haven't gotten to yet, but I'm working on it. Hoping to have a resolution for this ASAP.

**Comment by AnthonyMDev** [comment]
Actually, with the other one resolved, I was able to figure out what was wrong here and fix it super quickly! The fix for this is also no in the #3168 PR. Once that is merged in, we will push a patch and this should all be sorted out! So sorry it took so long to make this happen!
